packages/components/button/src/ButtonGroup/ButtonGroup.tsx

```diff
@@ -39,17 +39,24 @@ function _ButtonGroup(
       ref={ref}
       className={cx(styles.buttonGroup, className)}
     >
-      {React.Children.map(children, (child, key) => {
+      {React.Children.map(children, (child, index) => {
         if (!child) {
           return null;
         }
-        return React.cloneElement(child as React.ReactElement, {
-          key,
-          className: cx(
-            styles.groupContent,
-            (child as React.ReactElement).props.className,
-          ),
-        });
+        return (
+          <>
+            {withDivider && index !== 0 ? (
+              <div className={styles.divider} />
```

Comment:
❗️To be honest I am not a fan of this approach. As it adds multiple html-elements only to provide a little line. 

I am happy to hear what the others think, but I would preferably use a pseudo element instead of adding new divs.

Reply:
i tried pseudo elements first, the problem is that button has `overflow: hidden` styles and the border, which doesnt allow to make the divider full height.
happy to try other approaches if you have any suggestions

Reply:
<img width="840" alt="image" src="https://github.com/contentful/forma-36/assets/22265863/85e69ece-fc21-476f-869e-c244d9e8142d">


i also tried overriding border-right/left, but as you can see button has border-top/bottom as well, what make the divider cut in corners


<img width="977" alt="image" src="https://github.com/contentful/forma-36/assets/22265863/e4b7d50b-5aba-4652-b461-16bb3d9340bf">

and i tried to remove the border completely, and the result is visually what we want, but it makes button 2px shorter in width then it used to be

Reply:
hm, i think i can remove top/bottom border only, so it will keep the same with

Reply:
updated (including PR description)